app/controllers/employment/CheckEmploymentDetailsController.scala

```diff
@@ -48,7 +48,7 @@ class CheckEmploymentDetailsController @Inject()(implicit val cc: MessagesContro
                                                  clock: Clock
                                                 ) extends FrontendController(cc) with I18nSupport with SessionHelper with Logging {
 
-  def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = (authAction andThen TaxYearAction.taxYearAction(taxYear)).async { implicit user =>
```

Comment:
We should probably create a composed action from the authAction and the taxYearAction and make it usable through an object. This is a common code that should be used from within all the controllers and it is good to have one early.